drivers/timer/stm32_lptim_timer.c

```diff
@@ -61,11 +61,60 @@ static const struct device *const clk_ctrl = DEVICE_DT_GET(STM32_CLOCK_CONTROL_N
  *    0xFFFF / (LSE freq (32768Hz) / 128)
  */
 
-static uint32_t lptim_clock_freq = KHZ(32);
 static int32_t lptim_time_base;
 
 /* The prescaler given by the DTS and to apply to the lptim_clock_freq */
-#define LPTIM_CLOCK_RATIO DT_PROP(DT_DRV_INST(0), st_prescaler)
+#define LPTIM_PRESCALER DT_PROP(DT_DRV_INST(0), st_prescaler)
+
+#if (CONFIG_STM32_LPTIM_CLOCK_LSI)
+/*
+ * Most of the STM32_SERIES have a LSI @32000Hz,
+ * this frequency might differ on some that do not yet use LPTIM
+ */
+static uint32_t lptim_clock_freq = KHZ(32); /* Force STM32_SRC_LSI */
+
+/* Kconfig defines the clock source as LSI : check coherency with DTS */
+#if (DT_INST_CLOCKS_CELL_BY_IDX(0, 1, bus) != STM32_SRC_LSI)
+#warning CONFIG_STM32_LPTIM_CLOCK_LSI requires STM32_SRC_LSI defined as LPTIM domain clock
+#endif /* STM32_SRC_LSI */
+
+/* Kconfig is setting CONFIG_SYS_CLOCK_TICKS_PER_SEC adapted to the clock prescaler */
+#if !defined(CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE)
+/* For prescaler = 1,2,4,8 CONFIG_SYS_CLOCK_TICKS_PER_SEC <= 4000 is advised */
+#if ((CONFIG_SYS_CLOCK_TICKS_PER_SEC > 32000/8) && \
+	(LPTIM_PRESCALER < 16))
+#warning It is recommended to set SYS_CLOCK_TICKS_PER_SEC to 4000
+#endif
+/* For prescaler >= 16, set CONFIG_SYS_CLOCK_TICKS_PER_SEC <= 32000/prescaler */
+#if ((CONFIG_SYS_CLOCK_TICKS_PER_SEC > 32000/LPTIM_PRESCALER) && \
+	(LPTIM_PRESCALER >= 16))
+#warning It is recommended to set SYS_CLOCK_TICKS_PER_SEC to 32000/prescaler
+#endif
+#endif /* !CONFIG_STM32_LPTIM_TICK_FREQ_RATIO_OVERRIDE */
+
+#elif (CONFIG_STM32_LPTIM_CLOCK_LSE)
+static uint32_t lptim_clock_freq = 32768; /* Force STM32_SRC_LSE */
```

Comment:
Is that required ?

Reply:
` static uint32_t lptim_clock_freq = STM32_LPTIM_CLOCK`